packages/smart-vaults/src/utils/OperatorManager.sol

```diff
@@ -0,0 +1,199 @@
+    event OperatorRemoved(address indexed operator);
+
+    /// @notice Event emitted when an operator executes a call.
+    event ExecutedTxFromOperator(address indexed operator, Call call);
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   ERRORS                                   */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice Thrown when caller is not an operator.
+    error OnlyOperator();
+
+    /// @notice Thrown when target of call is this account.
+    error TargetCannotBeAccount();
+
+    /* -------------------------------------------------------------------------- */
+    /*                                  MODIFIERS                                 */
+    /* -------------------------------------------------------------------------- */
+
+    modifier onlyOperator() {
+        if (!_getOperatorManagerStorage().isOperator[msg.sender]) revert OnlyOperator();
+        _;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                          EXTERNAL/PUBLIC FUNCTIONS                         */
+    /* -------------------------------------------------------------------------- */
+
+    /**
+     * @notice Adds operator to the allowlist.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     */
+    function addOperator(address operator_) public {
+        _authorize();
+
+        _addOperator(operator_);
+    }
+
+    /**
+     * @notice Adds `operator` to the allowlist and makes a call to the `setupContract` passing `data_`.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     * @param setupContract_ address of contract to call to setup operator.
+     * @param data_ data passed to the setupContract.
+     */
+    function addAndSetupOperator(address operator_, address setupContract_, bytes calldata data_) public {
+        _authorize();
+
+        _addOperator(operator_);
+
+        _call(setupContract_, 0, data_);
+    }
+
+    /**
+     * @notice Removes operator from the allowlist.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     */
+    function removeOperator(address operator_) public {
+        _authorize();
+
+        _removeOperator(operator_);
+    }
+
+    /**
+     * @notice Removes `operator` from the allowlist and makes a call to the `teardownContract_` passing `data_`.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     * @param teardownContract_ address of contract to call to teardown operator.
+     * @param data_ data passed to the setupContract.
+     */
+    function removeAndTeardownOperator(address operator_, address teardownContract_, bytes calldata data_) public {
```

Comment:
`install` / `uninstall` might be better nomenclature idk

Reply:
hmm, decided to not use install/uninstall initially because its used in modules and thought it might be confusing but don't feel strongly about it